src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -154,8 +155,10 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
                     }
                 }
             }
-            this.connectionString.username = username;
-            this.connectionString.password = password;
+            if (username && password) {
```

Comment:
This check shouldn't be necessary

Reply:
We still need this check for azure accounts as invalidCredentialsErrorType would lead to `enterPostgresCredentials()` when trying to load contents of a database. This for only when the account is created in the portal and not in the extension.

Reply:
Can you explain this code path? I'm still a bit confused. It seems like if "username" and "password" are undefined (aka not persisted or in keytar), then `this.connectionString.username` and `this.connectionString.password` would also be undefined and so it wouldn't matter if you have this check or not

Reply:
You're right, the check here is not necessary